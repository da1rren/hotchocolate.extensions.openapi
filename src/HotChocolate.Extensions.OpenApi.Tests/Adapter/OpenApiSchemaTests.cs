namespace HotChocolate.Extensions.OpenApi.Tests.Adapter;

using OpenApi.Adapter;
using Snapshooter.Xunit;

public class OpenApiSchemaTests
{
    [Fact]
    public void OpenApiField_Should_Produce_Valid_Graphql_Field()
    {
        var field = new OpenApiTypeField("foo", "Int");
        Snapshot.Match(field.ToString());
    }

    [Fact]
    public void OpenApiType_Should_Produce_Valid_Graphql_Type()
    {
        var field = new OpenApiTypeField("foo", "Int");
        var type = new OpenApiType("bar", field);
        Snapshot.Match(type.ToString());
    }

    [Fact]
    public void OpenApiSchema_Should_Produce_Valid_Graphql_Schema()
    {
        var field = new OpenApiTypeField("foo", "Int");
        var barType = new OpenApiType("bar", field);
        var fooType = new OpenApiType("foobar", field);
        var schema = new OpenApiSchema(new[] {barType, fooType}, Array.Empty<IOpenApiOperation>())
            .ToString();

        Snapshot.Match(schema);
    }
}