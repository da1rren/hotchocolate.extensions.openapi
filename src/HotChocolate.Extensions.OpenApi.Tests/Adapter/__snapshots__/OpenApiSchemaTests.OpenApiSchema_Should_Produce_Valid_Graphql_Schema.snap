
type bar { 
    foo: Int
}

type foobar { 
    foo: Int
}
